File: source/downloads/code/Fibonacci-Number-in-Different-Programming-Languages/Python/fibonacci.py
import sys


def fib(n):
    if n <= 1:
        return n

    a = 0
    b = 1

    for i in range(2, n + 1):
        c = a + b
        a = b
        b = c

    return b


if __name__ == '__main__':
    n = int(sys.argv[1])
    print(f"fib({n}) = {fib(n)}\n")

File: source/downloads/code/Fibonacci-Number-in-Different-Programming-Languages/C++/fibonacci.cpp
#include <iostream>

using namespace std;

int fib(int n) {
    if(n <= 1) {
        return n;
    }

    int a = 0;
    int b = 1;

    for(int i = 2; i <= n; i++) {
        int c = a + b;
        a = b;
        b = c;
    }
    return b;
}

int main(int argc, char *argv[]) {
    int n = atoi(argv[1]);
    cout << "fib(" << n << ")" << " = " <<  fib(n) << endl;
    return 0;
}
